pyocd/target/builtin/target_AMA3B1KK.py

```diff
@@ -0,0 +1,211 @@
+    'pc_eraseAll': 0x10000037,
+
+    'static_base' : 0x10000000 + 0x00000020 + 0x000000f4,
+    'begin_stack' : 0x10000400,
+    'begin_data' : 0x10000000 + 0x1000,
+    'page_size' : 0x2000,
+    'analyzer_supported' : False,
+    'analyzer_address' : 0x00000000,
+    'page_buffers' : [0x10001000, 0x10003000],   # Enable double buffering
+    'min_program_length' : 0x2000,
+
+    # Flash information
+    'flash_start': 0xc000,
+    'flash_size': 0xF4000,
+    'sector_sizes': (
+        (0x0, 0x2000),
+    )
+}
+
+class Apollo3(CortexM):
+    MCUCTRL_SCRATCH0 = 0x400401B0
+    MCUCTRL_BOOTLDR = 0x400401A0
+    JDEC_PID = 0xF0000FE0
+
+    def __init__(self, session, ap, memoryMap, core_num, acquire_timeout):
+        self._acquire_timeout = acquire_timeout
+        super(Apollo3, self).__init__(session, ap, memoryMap, core_num)
+
+    def set_reset_catch(self, reset_type=None):
+        """! @brief Prepare to halt core on reset."""
+        #LOG.debug("set reset catch, core %d", self.core_number)
+
+        self._reset_catch_delegate_result = self.call_delegate('set_reset_catch', core=self, reset_type=reset_type)
+        
+        # Default behaviour if the delegate didn't handle it.
+        if not self._reset_catch_delegate_result:
+            # Halt the target.
+            self.halt()
+
+        jdecpid = self.read_memory(self.JDEC_PID)
+        #print("JDEC PID " + str(jdecpid))
+
+        if((jdecpid & 0xf0) == 0xc0):
+            #print("Ambiq Apollo3")
+            bootldr = self.read_memory(self.MCUCTRL_BOOTLDR)
+            if ((bootldr & 0x0C000000) == 0x04000000):
+                #print("Secure Part.")
+                secure = True
+        
+        if(secure):
+            scratch0 = self.read_memory(self.MCUCTRL_SCRATCH0)
+            #print("scratch0 = "+ str(scratch0))
+            self.write_memory(self.MCUCTRL_SCRATCH0, (scratch0 | 0x1))
+            #print("wrote " + str(scratch0 | 0x1) + " to scratch0")
+        else:
+            super(Apollo3, self).set_reset_catch(reset_type)
+            #print("I havent wrtten non secure yet")
+
+    def reset(self, reset_type=None):
+        super(Apollo3, self).reset(reset_type)
+        sleep(0.1)
+
+class AMA3B1KK(CoreSightTarget):
+    
+    VENDOR = "Ambiq Micro"
+    CortexM_Core = Apollo3
+
+    memoryMap = MemoryMap(
+        FlashRegion(start=0x0000c000, length=0x000F4000, sector_size=0x2000,
+                        page_size=0x200,
+                        is_boot_memory=True,
+                        algo=flash_algo),
+        RamRegion(  start=0x10000000,  length=0x60000)
+        )
+
+    def __init__(self, link):
+        self.DEFAULT_ACQUIRE_TIMEOUT = 25.0
+        super(AMA3B1KK, self).__init__(link, self.memoryMap)
+        self._svd_location = SVDFile.from_builtin("apollo3.svd")
+
+    def create_init_sequence(self):
+        seq = super(AMA3B1KK, self).create_init_sequence()
+        seq.replace_task('create_cores', self.create_ap3_core)
```

Comment:
This won't work on master. The init sequence has been modified to use a `CoreSightDiscovery` class that now contains the 'create_cores' task. You can see updated code similar to this here:
 https://github.com/mbedmicro/pyOCD/blob/ed54d2689410d2ba0051b56265aa41a37343d395/pyocd/target/family/target_lpc5500.py#L58

Probably you want to rebase onto the HEAD of master.

Reply:
thanks for this information, completed.